### PR TITLE
Preserve trivia when converting a numeric literal base

### LINKED ISSUE #19225: Convert to binary/decimal/hex deletes leading trivia sometimes
```C#
class Program
{
    void M()
    {
        var numbers = new int[] {
            0x1, 0x2
        };
    }
}
```

If you invoke `Convert to binary` on the `0x1`, it deletes all the leading trivia.  This doesn't happen for `0x2`, or if `0x1` is the only item in the initializer.

### PR DESCRIPTION
**Pull request by alrz**
Fixes #19225

### CODE REVIEW

**Comment by Pilchie** [comment]
Tagging @dotnet/roslyn-ide for some reviews.

File: src/EditorFeatures/VisualBasicTest/CodeActions/ConvertNumericLiteral/ConvertNumericLiteralTests.vb [hunk 1]
```diff
@@ -89,5 +89,27 @@ End Class"
...
+    End Sub
+End Class",
+"Class X
+    Sub M()
+        Dim x As Integer() =
+        {
+            &B1, &H2
+        }
+    End Sub
+End Class", index := Refactoring.ChangeBase2, ignoreTrivia := False)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/CodeActions/ConvertNumericLiteral/ConvertNumericLiteralTests.vb]: nit: no spaces around :=

**Comment by sharwell** [comment]
@MattGertz for ask mode

## Ask Mode

**Customer scenario**

A customer uses our refactoring to convert a numeric literal from one base to another, and loses trivia (comments, preprocessor directives, and/or formatting) associated with it.

**Bugs this fixes:**

#19225

**Workarounds, if any**

Convert the number by hand.

**Risk**

Very low. The refactoring operation consumes exactly one token input and produces exactly one token output. Preservation of trivia during this transformation is safe.

**Performance impact**

Negligible. There is no additional overhead except in the case where the previous code behaved incorrectly, and the solution provides optimal performance for that case.

**Is this a regression from a previous update?**

No.

**Root cause analysis:**

Failure to test trivia cases for refactoring features.

**How was the bug found?**

Customer reported.

**Comment by Pilchie** [comment]
Thanks @alrz!

**Pull request merged**